engine/src/main/java/org/hibernate/search/util/impl/ReflectionHelper.java

```diff
@@ -115,13 +115,34 @@ public static Object getMemberValue(Object bean, XMember getter) {
 	}
 
 	/**
-	 * Checks whether the specified class contains any Search specific annotations.
+	 * Checks whether the specified class contains any Search-specific annotations, i.e. if
+	 * the class itself or one of its superclasses carries such an annotation.
 	 *
 	 * @param mappedClass the {@code XClass} to check for Search annotations
 	 *
 	 * @return Returns {@code true} if the class contains at least one Search annotation, {@code false} otherwise
 	 */
 	public static boolean containsSearchAnnotations(XClass mappedClass) {
+		List<XClass> hierarchy = ReflectionHelper.createXClassHierarchy( mappedClass );
+
+		for ( XClass clazz : hierarchy ) {
```

Comment:
I agree it would be weird (total nonsense, even) to index something on `Object`, at least in production. I guess we could imagine someone indexing `toString` for debugging purposes in a development environment. Though it'd still be weird.

Anyway, I don't think the number of classes inspected here is that important. We're talking about one execution per class in the hierarchy of each non-`@Indexed` Hibernate classes, and only when boostrapping the Search factory. Which is really not much work, especially since it does not involve I/O.

If we were talking about runtime code, why not, but in this case I find the weirdness of excluding `Object` more annoying than the performance hit (!) of inspecting the `Object` class.